packages/tbds-app-frame/lib/app-frame.scss

```diff
@@ -0,0 +1,26 @@
+.app-frame {
+  align-items: stretch;
+  display: flex;
+  flex-direction: column;
+  min-height: 100vh;
+  width: 100%;
+}
+
+.app-frame__header {
+  flex-grow: 0;
+}
+
+.app-frame__body {
+  flex-grow: 1;
+}
+
+.app-frame__footer {
+  flex-grow: 0;
+}
+
+.app-frame__body--vertical-middle {
```

Comment:
I wonder if this could be a different pattern. I looks like it is designed to align things on a vertical line within the body? 

Or were you intending to do an `align-self` instead?

Reply:
Good question. This modifier came from needing to vertically aligned the body contents. A good example of this is login pages, or pages that have a vertically aligned "dialog" element. Here it is used by tbot:

![screen shot 2018-03-16 at 14 05 30](https://user-images.githubusercontent.com/903327/37537175-1c3d52aa-2923-11e8-8fa9-da1d9e85483f.png)

`align-self` would work, but you'd need to know of and apply that to the thing being vertically centered. With the dialog example, I would rather the dialog be completely agnostic of how it placed within its parent, and instead let the parent (`app-frame`) lay it out.

Maybe we need a better name for this modifier?

Reply:
Ahh I was thinking you were getting the `body` to align to the center and keep the header and footer on the edges. That way it wouldn't be interacting with any children of the `body`. 

By asking the browser to coerce the children to be come `flex:` items, the developer would just need to know that it should be a single child. Seems fine to me. Once we get the markup examples in, it will become real clear.

Reply:
That's a great point and would make for some fantastic docs!